history/taskana-simplehistory-provider/src/test/java/acceptance/events/CreateHistoryEventOnTaskCancellationAccTest.java

```diff
@@ -0,0 +1,45 @@
+package acceptance.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import acceptance.AbstractAccTest;
+import acceptance.security.JaasExtension;
+import acceptance.security.WithAccessId;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import pro.taskana.simplehistory.impl.HistoryEventImpl;
+import pro.taskana.simplehistory.impl.SimpleHistoryServiceImpl;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.TaskState;
+import pro.taskana.task.api.models.Task;
+
+@ExtendWith(JaasExtension.class)
+class CreateHistoryEventOnTaskCancellationAccTest extends AbstractAccTest {
+
+  private final TaskService taskService = taskanaEngine.getTaskService();
+  private final SimpleHistoryServiceImpl historyService = getHistoryService();
+
+  @Test
+  @WithAccessId(user = "admin")
+  void should_CreateCancelledHistoryEvent_When_TaskIsCancelled() throws Exception {
+
+    final String taskId = "TKI:000000000000000000000000000000000001";
+
+    List<HistoryEventImpl> listEvents = historyService.createHistoryQuery().taskIdIn(taskId).list();
+
+    assertThat(listEvents).isEmpty();
+
+    assertThat(taskService.getTask(taskId).getState()).isEqualTo(TaskState.CLAIMED);
```

Comment:
I don't know what the good practice is. I personally prefer to separate action from assertion to make it more readable. By maybe Sonar complains? 
What do you think?

Reply:
BTW: do we need the task to be in state CLAIMED to cancel it?

Reply:
Sonar is not complaining. Personally I think as long as the assert fits into 1 LOC the readability should be granted. For me this is similar to returning something from a method. You can instantly return something or declare it as a variable before returning. Sonar tends to not like creating extra variables before returning. What do you think @mustaphazorgati ? 

No the task state can be anything as long as it is not an end state. I could probably remove the check for the task state before terminating the task if you want me to.

Reply:
Maybe we should discuss it in the CoP meeting?
Regarding the status: in this case we do not neet the claim and the check for it. But we should have tests for both cancel operations: one from ready state and one from claimed state.